### ISSUE TITLE
Fix DatetimeIndex and other docstring for missing period at the end of summary

### ISSUE THREAD

**Issue by datapythonista** [python-sprints/pandas-mentoring#138]
Same as #132 but for the next docstrings:
```
pandas.IndexSlice: Summary does not end with a period
pandas.MultiIndex.names: Summary does not end with a period
pandas.MultiIndex.is_lexsorted: Summary does not end with a period
pandas.MultiIndex.reorder_levels: Summary does not end with a period
pandas.DatetimeIndex.snap: Summary does not end with a period
pandas.DatetimeIndex.to_perioddelta: Summary does not end with a period
pandas.DatetimeIndex.to_pydatetime: Summary does not end with a period
pandas.DatetimeIndex.to_series: Summary does not end with a period
pandas.TimedeltaIndex: Summary does not end with a period
pandas.PeriodIndex.is_leap_year: Summary does not end with a period
pandas.api.extensions.ExtensionArray._concat_same_type: Summary does not end with a period
pandas.api.extensions.ExtensionArray.dropna: Summary does not end with a period
```

**Comment by martinagvilas** [comment]
I would like to work on this

**Comment by martinagvilas** [comment]
Closing this issue because it was merged in pandas #28031
